src/app/clusters/ota-requestor/README.md

```diff
@@ -3,48 +3,131 @@
+    confirmed, the provider is immediately notified of the update being applied
+    successfully
+-   If an existing OTA update is already in progress, any new attempts to query
+    will be denied
+-   A periodic query is attempted every 24 hours, unless overridden
+-   The periodic query timer starts whenever `OTARequestorInterface` is in the
+    idle state
 
-## Design Overview
+### OTAImageProcessorInterface
+
+This is a platform-agnostic interface for processing downloaded chunks of OTA
+image data. The data could be raw image data meant for flash or metadata. This
+component should interact with the `OTADownloader` to drive the download
+process.
+
+Each platform should provide an implementation of this interface which should
+reside under `src/platform/<platform-name>`.
+
+### OTADownloader
+
+This is an interface for image download functionality over a particular
+protocol. Each `DownloadProtocolEnum` supported should provide an implementation
+of this interface.
 
-To be completed..
+`BDXDownloader` class is an implementation of this interface for the BDX
+protocol.
+
+### OTARequestorStorage
+
+This is an interface for storing/loading persistent data related to OTA.
+
+`DefaultOTARequestorStorage` class is the default implementation of this
+interface. Any custom implementation should reside under
+`examples/platform/<platform-name>`.
+
+## Steps for including the OTA Requestor functionality in a Matter application
+
+-   Enable `Server` for the OTA Software Update Requestor cluster in the
+    application zap file
+-   Enable `Client` for the OTA Software Update Provider cluster in the
+    application zap file
+-   Implement OTA Requestor components:
+    -   Use the `DefaultOTARequestor` class or implement a class derived from
+        `OTARequestorInterface`
+    -   Use the `DefaultOTARequestorDriver` class or implement a class derived
+        from `OTARequestorDriver`
+    -   Use the `BDXDownloader` class or implement a class derived from
+        `OTADownloader`
+    -   Implement a class derived from `OTAImageProcessorInterface`
+    -   Use the `DefaultOTARequestorStorage` class or implement a class derived
+        from `OTARequestorStorage`
+-   If using the default implementation of the interfaces defined above,
+    explicitly list all the source files in `src/app/clusters/ota-requestor` in
+    the application make/build file. For example: `src/app/chip_data_model.gni`.
+    Otherwise, list the source files where the component implementation reside.
+-   Explicitly list all the OTA platform specific files in `src/platform`. For
+    example: `src/platform/Linux/BUILD.gn`
+-   Instantiate and initialize each component in the application. For example,
+    in an application which uses the default implementation on the Linux
+    platform:
+    -   Create an instance of the `DefaultOTARequestor` class
+    -   Create an instance of the `DefaultOTARequestorDriver` class
```

Comment:
I think the driver should be initialized as the last one. Maybe it would be good to define these in the order which is currently used by the examples to avoid surprises :)

Reply:
I think the only dependency is that the `DefaultOTARequestor::Init` is called before `DefaultOTARequestorDriver::Init` since we could potentially go down a path of calling mRequestor->Reset. In either case, I've moved the connecting documentation around such that it matches the one in Linux/main.cpp and called out that the driver must be initialized after the `DefaultOTARequestor::Init`